components/motor/roboclaw/roboclaw.go

```diff
@@ -158,13 +174,48 @@ func (m *roboclawMotor) SetPower(ctx context.Context, powerPct float64, extra ma
 	}
 }
 
+func goForMath(rpm, revolutions float64) (float64, time.Duration) {
+	// If revolutions is 0, the returned wait duration will be 0 representing that
+	// the motor should run indefinitely.
+	if revolutions == 0 {
+		powerPct := 1.0
+		return powerPct, 0
+	}
+
+	dir := rpm * revolutions / math.Abs(revolutions*rpm)
+	powerPct := math.Abs(rpm) / maxRPM * dir
+	waitDur := time.Duration(math.Abs(revolutions/rpm)*60*1000) * time.Millisecond
```

Comment:
[nit] move constants to top of file, so you conversion factor of 60s to milliseconds